src/shared/s3_manager.py

```diff
@@ -64,6 +75,18 @@ def __init__(self, event):
                 f"{self.study}__{self.data_package}__{self.site}__{self.version}/"
                 f"{self.study}__{self.data_package}__{self.site}__flat.parquet"
             )
+        if study:
+            self.study = study
+        if site:
+            self.site = site
+        if data_package:
+            self.data_package = data_package
+        if version:
+            self.version = version
```

Comment:
This block struck me - the  `if "Records" in event` block will set these variables too and use them to define other variables. So by setting these afterward, you now would have inconsistent state, yah? Like `self.parquet_aggregate_path` won't match these variables if they are passed in.

Would it make sense to move these above and then also drop the conditionals, so you just set these class properties from the args unconditionally, and then maybe the even overrides them? Or is the current ordering intentional?